tests/agents/microchain/test_messages_functions.py

```diff
@@ -0,0 +1,148 @@
+import os
+import typing
+from typing import Generator
+from unittest.mock import PropertyMock, patch
+
+import polars as pl
+import pytest
+from eth_typing import ChecksumAddress
+from prediction_market_agent_tooling.gtypes import xdai_type
+from prediction_market_agent_tooling.tools.utils import check_not_none
+from prediction_market_agent_tooling.tools.web3_utils import xdai_to_wei
+from pydantic import SecretStr
+from web3 import Web3
+
+from prediction_market_agent.agents.microchain_agent.messages_functions import (
+    ReceiveMessage,
+)
+from prediction_market_agent.agents.microchain_agent.utils import compress_message
+from prediction_market_agent.db.blockchain_transaction_fetcher import (
+    BlockchainTransactionFetcher,
+)
+from prediction_market_agent.utils import DBKeys
+
+
+@pytest.fixture(scope="module")
+def agent2_address() -> ChecksumAddress:
+    return Web3.to_checksum_address("0xb4D8C8BedE2E49b08d2A22485f72fA516116FE7F")
+
+
+MOCK_HASH_1 = "mock_hash1"
+MOCK_HASH_2 = "mock_hash2"
+
+
+def mock_spice_query(query: str, api_key: str) -> pl.DataFrame:
+    anvil_account_1 = Web3.to_checksum_address(
+        "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
+    )
+    compress_message("hello hello hello")
+    return pl.DataFrame(
+        {
+            "hash": [MOCK_HASH_1, MOCK_HASH_2],
+            "value": [xdai_to_wei(xdai_type(1)), xdai_to_wei(xdai_type(2))],
+            "block_number": [1, 2],
+            "from": [anvil_account_1, anvil_account_1],
+            "data": ["test", Web3.to_hex(compress_message("test"))],
+        }
+    )
+
+
+@pytest.fixture(scope="module")
+def patch_dune_api_key() -> Generator[PropertyMock, None, None]:
+    with patch(
+        "prediction_market_agent.utils.APIKeys.dune_api_key",
+        new_callable=PropertyMock,
+    ) as mock_dune:
+        mock_dune.return_value = SecretStr("mock_dune_api_key")
+        yield mock_dune
+
+
+@pytest.fixture(scope="module")
+def patch_spice() -> Generator[PropertyMock, None, None]:
+    with patch(
+        "spice.query",
+        side_effect=mock_spice_query,
+    ) as mock_spice:
+        yield mock_spice
+
+
+@pytest.fixture
+def patch_public_key(
+    agent2_address: ChecksumAddress,
+) -> Generator[PropertyMock, None, None]:
+    with patch(
+        "prediction_market_agent.agents.microchain_agent.microchain_agent_keys.MicrochainAgentKeys.public_key",
+        new_callable=PropertyMock,
+    ) as mock_public_key:
+        mock_public_key.return_value = agent2_address
+        yield mock_public_key
+
+
+@pytest.fixture
+def patch_pytest_db(
+    session_keys_with_mocked_db: DBKeys,
+) -> Generator[PropertyMock, None, None]:
+    # Mocking os.environ because mocking Pydantic's attribute is not possible via patch.
```

Comment:
Then how come you are doing it in the patches above? 🤔

Reply:
My mistake. Refactored to a similar logic as above.